src/lib/components/Sidebar.svelte

```diff
@@ -4,14 +4,88 @@
   import type { UserWithUserTeamsActiveTeamAndChats } from '$lib/server/entities/user'
   import { isSidebarOpen } from '$lib/stores/general'
   import { ChevronRightIcon, PlusIcon } from '@babeard/svelte-heroicons/solid'
+  import {
+    getTimeSince,
+    isToday,
+    isLastYear,
+    isYesterday,
+    millisecondsPerDay,
+    previousSevenDays,
+    lastThirtyDays,
+    currentYear,
+  } from '$lib/utils/time'
 
   export let user: UserWithUserTeamsActiveTeamAndChats
+
+  const renderedTitles: { [key: string]: boolean } = {}
+
   $: chats = [
     ...(user?.sharedChats
       .filter((x) => x.chat.owner.teamId == user.activeUserTeam?.teamId)
       .map((x) => x.chat) || []),
     ...(user.activeUserTeam?.chats || []),
   ].sort((a, b) => b.updatedAt.getTime() - a.updatedAt.getTime())
+
+  type ChatLists = {
+    [key: string]: any[]
+    today: any[]
+    yesterday: any[]
+    previousSevenDays: any[]
+    lastMonth: any[]
+  }
+
+  $: chatsGroupedByTime = chats?.reduce(
+    (chatLists: ChatLists, chat) => {
+      if (isToday(chat.updatedAt)) {
+        //!chatLists['today'] ? (chatLists['today'] = []) : chatLists['today'].push(chat)
+        chatLists.today.push(chat)
+      } else if (isYesterday(chat.updatedAt)) {
+        chatLists.yesterday.push(chat)
+      } else if (
+        millisecondsPerDay < getTimeSince(chat.updatedAt) &&
+        getTimeSince(chat.updatedAt) <= previousSevenDays
+      ) {
+        chatLists.previousSevenDays.push(chat)
+      } else if (
+        previousSevenDays < getTimeSince(chat.updatedAt) &&
+        getTimeSince(chat.updatedAt) <= lastThirtyDays
+      ) {
+        chatLists.lastMonth.push(chat)
+      } else if (chat.updatedAt.getFullYear() >= currentYear) {
+        const year = chat.updatedAt.getFullYear()
+        if (!chatLists[year]) {
+          chatLists[year] = []
+        }
+        chatLists[year].push(chat)
+      }
```

Comment:
I think you could make this more readable by putting all of the `if`-statements inside a new function in `time.ts`: something like `categorizeDate(date: Date): string`. Than function would run a date through the `if`-statements and return a classification ("today", "lastMonth", "2021" etc.). Then this code would just be:
```js
const category = categorizeDate(chat.updatedAt)
if (!chatLists[category]) {
  chatLists[category] = []
}
chatLists[category].push(chat)
```

Reply:
This would also have the positive side-effect that your categorization code would be testable 😊